### ISSUE TITLE
[Question] Is it expected that LH & storage still keeps volume config even volume backup is deleted on Harvester?

### ISSUE THREAD

**Issue by albinsun** [harvester/harvester#4529]
## Question
After delete the (only) VM backup on Harvester, we found that
1. Longhorn still keep the backing image record
2. Storage still keep `volume.cfg`

Is this behavior by design?

## Environment
* Harvester
  * Version: `v1.2.0`
  * Profile: QEMU/KVM, single node (8C/16G/250G)
  * ui-source: `Auto`

## Reproduce Steps
1. Fresh install Harvester and empty NFS storage

1. Setup `backup-target`
   <details>
   ![image](https://github.com/harvester/harvester/assets/2773781/99df45d1-cb6f-4fc3-96f4-348d833adfea)
   </details>
   
1. Create VM & Take backup
   ![image](https://github.com/harvester/harvester/assets/2773781/5f93ffd3-040b-4401-95ce-2bbadc36a531)

1. Delete VM Backup from Harvester
   * blocks are deleted
   * metadata still keep on LH and storage
   ![image](https://github.com/harvester/harvester/assets/2773781/18e53afa-b085-417a-a66f-66e2f57adf76)

## Note
1. Files can be complete cleanup via delete backup record from Longhorn
   * Before
     ![image](https://github.com/harvester/harvester/assets/2773781/ab6b5055-e0b2-467f-a67c-b4f8b3bdb1bb)
   * After
     ![image](https://github.com/harvester/harvester/assets/2773781/e963a3d1-a4b2-4694-ba8b-39a312ed54b4)

**Comment by WebberHuang1118** [comment]
TL;DR: Yes, it's expected by LH's implementation.

There are two key CR for volume backup: `Backup` and `BackupVolume`
- `Backup` represents an LH snapshot backup (Listed in LH GUI `Backup -> specific volume`)
- `BackupVolume` represents a set containing all backups for a specific LH volume (Listed in LH GUI `Backup`)

As a volume backup is triggered, the backupstore data usage increases in the match data path (E.q. for NFS backup is `<NFS path>/backupstore/volumes/xx/xx/<volume>/`). In contrast, as a backup is deleted, the backupstore data usage decreases in the match data path.

Once all backups are gone for a specific volume, there is only a `volume.cfg` left in the backupstore. And `volume.cfg`'s size is small (< 1MB). 

Maybe there is room to improve, for example, once there is no backup for a certain volume, delete all the data on backupstore (i.e. delete CR `BackupVolume`). 

However, considering the size of the left data is not great, IMHO, it should not have a big impact on users so far. Thanks.

**Comment by albinsun** [comment]
Thanks for checking, will discuss about.
